### PR TITLE
Task/bws15 master sign up

### PR DESCRIPTION
**Pull request by junojei**

### CODE REVIEW

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+@import url("https://fonts.googleapis.com/css?family=Montserrat");
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.scss]: connect styles in the `index.html`

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-import { ButtonType, InputType } from 'common/enums/enums';
+import logo from 'assets/img/logo.svg';
+import { AppRoute, ButtonType, InputType } from 'common/enums/enums';
 import { useAppForm } from 'hooks/hooks';
 import { getNameOf } from 'helpers/helpers';
 import { MasterSignUpDto } from 'common/types/types';
-import { Button, Input } from 'components/common/common';
+import { Button, Input, Link } from 'components/common/common';
 import { DEFAULT_REGISTER_PAYLOAD } from './common/constants';
+import './sign-up-form.scss';
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]: we should use `styles-modules`

File: shared/src/common/enums/validation/master/master-sign-up.validation-message.ts [hunk 3]
```diff
@@ -1,6 +1,11 @@
 const MasterSignUpValidationMessage = {
   EMAIL_REQUIRE: 'Email is required',
   EMAIL_WRONG: 'Email is wrong',
+  NAME_REQUIRE: 'Name is required',
+  NAME_MIN_LENGTH: 'Name must have at least 4 characters',
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/master/master-sign-up.validation-message.ts]: create enum `MasterSignUpValidationRule` and use here (for all statements)

```suggestion
  NAME_MIN_LENGTH: 'Name must have at least ${MasterSignUpValidationRule. NAME_MIN_LENGTH} characters',
```

File: shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts [hunk 4]
```diff
@@ -12,8 +12,24 @@ const masterSignUp = Joi.object({
       'string.email': MasterSignUpValidationMessage.EMAIL_WRONG,
       'string.empty': MasterSignUpValidationMessage.EMAIL_REQUIRE,
     }),
-  [getNameOf<MasterSignUpDto>('name')]: Joi.string().trim().required(),
-  [getNameOf<MasterSignUpDto>('password')]: Joi.string().trim().required(),
+  [getNameOf<MasterSignUpDto>('name')]: Joi.string()
+    .trim()
+    .min(4)
```
Comment (what1s1ove) [inline shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts]:
```suggestion
    .min(MasterSignUpValidationRule.NAME_MIN_LENGTH)
```
avoid magic number. add it for all statements

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/common/enums/validation/master/master-sign-up.validation-message.ts [hunk 5]
```diff
@@ -1,6 +1,13 @@
+import { MasterSignUpValidationRule } from './master-sign-up.validation-rule';
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/master/master-sign-up.validation-message.ts]: rename file to `shared/src/common/enums/validation/master/master-sign-up-validation-message.enum.ts`

File: shared/src/common/enums/validation/master/master-sign-up.validation-rule.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const MasterSignUpValidationRule = {
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/master/master-sign-up.validation-rule.ts]: rename file to `shared/src/common/enums/validation/master/master-sign-up-validation-rule.enum.ts`

File: shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@ import * as Joi from 'joi';
 import { getNameOf } from '~/helpers/helpers';
 import { MasterSignUpDto } from '~/common/types/types';
 import { MasterSignUpValidationMessage } from '~/common/enums/enums';
+import { MasterSignUpValidationRule } from '~/common/enums/enums';
```
Comment (what1s1ove) [inline shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts]:
```suggestion
import { MasterSignUpValidationMessage, MasterSignUpValidationRule } from '~/common/enums/enums';
```

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx [hunk 8]
```diff
@@ -18,9 +20,19 @@ const SignUpForm: React.FC<Props> = ({ onSubmit }) => {
 
   return (
     <>
-      <h1>Sign Up ⚛️</h1>
-      <form onSubmit={handleSubmit(onSubmit)}>
-        <p>
+      <div className={styles['sign-up-logo']}>
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]:
```suggestion
      <div className={styles.signUpLogo}>
```

Comment (what1s1ove) [reply frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]: don't forget to change styles also

File: frontend/public/index.html [hunk 9]
```diff
@@ -11,6 +11,10 @@
     />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css?family=Montserrat"
+    />
```
Comment (what1s1ove) [inline frontend/public/index.html]:
```suggestion
  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;600;700&display=swap" />
```
don't forget about font-weight

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx (hunk 8 above)
Comment (junojei) [reply frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]: зображення
Lint is angry

Comment (what1s1ove) [reply frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]: pull last changes from the development 👍

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.module.scss [hunk 10]
```diff
@@ -0,0 +1,42 @@
+.signUpTitle,
+.signUpForm,
+.signInLink,
+.signUpFormPolicy,
+.signUpLogo {
+  width: 25%;
+  margin: 0 auto;
+  color: #a0a0a0;
+  font-family: Montserrat, sans-serif;
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.module.scss]:
```suggestion
  font-family: Montserrat, sans-serif;
```
just add it as a global

File: shared/src/common/enums/validation/master/master-sign-up-validation-rule.enum.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const MasterSignUpValidationRule = {
+  NAME_MIN_LENGTH: 4,
+  NAME_FIRST_CHARTER: /^[a-zA-Z].+$/,
+  PASSWORD_MIN_LENGTH: 6,
+};
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/master/master-sign-up-validation-rule.enum.ts]:
```suggestion
} as const;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]
👍

**Pull request merged**